app/models/responses/PropertyPeriodicSubmissionRequest.scala

```diff
@@ -16,72 +16,101 @@
+                                                   ukFhlProperty: Option[UkFhlProperty],
+                                                   ukOtherProperty: Option[UkOtherProperty])
 
 object PropertyPeriodicSubmissionRequest {
   implicit val format: OFormat[PropertyPeriodicSubmissionRequest] = Json.format[PropertyPeriodicSubmissionRequest]
 
-  def fromExpenses(expenses: Expenses): PropertyPeriodicSubmissionRequest = {
-    PropertyPeriodicSubmissionRequest(
-      None,
-      None,
-      None,
-      None,
-      Some(
-        UkOtherProperty(
-          UkOtherPropertyIncome(Some(0), None, None, None, None, None),
-          UkOtherPropertyExpenses(
-            premisesRunningCosts = expenses.rentsRatesAndInsurance,
-            repairsAndMaintenance = expenses.repairsAndMaintenanceCosts,
-            financialCosts = expenses.loanInterest,
-            professionalFees = expenses.otherProfessionalFee,
-            costOfServices = expenses.costsOfServicesProvided,
-            travelCosts = expenses.propertyBusinessTravelCost,
-            other = expenses.otherAllowablePropertyExpenses,
-            residentialFinancialCostsCarriedForward = None,
-            ukOtherRentARoom = None,
-            consolidatedExpense = None,
-            residentialFinancialCost = None
+  def fromExpenses(periodicSubmissionMaybe: Option[PropertyPeriodicSubmission], expenses: Expenses): Either[ServiceError, PropertyPeriodicSubmissionRequest] = {
+
+    val ukOtherPropertyIncomeWithPeriodicSubmission = for {
+      periodicSubmission <- periodicSubmissionMaybe
+      ukOtherPropertyIncome <- periodicSubmission.ukOtherProperty.map(_.income)
+    } yield (periodicSubmission, ukOtherPropertyIncome)
+
+    ukOtherPropertyIncomeWithPeriodicSubmission.fold[Either[ServiceError, PropertyPeriodicSubmissionRequest]](
+      //Todo: Maybe elaborate each
+      InternalError("No periodicSubmission / uk other property income is present, should be together with expenses").asLeft[PropertyPeriodicSubmissionRequest]
+    )(result => {
+
+      val (periodicSubmission, ukOtherPropertyIncome) = result
+      PropertyPeriodicSubmissionRequest(
+        periodicSubmission.submittedOn, //Todo: Change to now?
+        periodicSubmission.foreignFhlEea,
+        periodicSubmission.foreignProperty,
+        periodicSubmission.ukFhlProperty,
+        Some(
+          UkOtherProperty(
+            ukOtherPropertyIncome,
+            UkOtherPropertyExpenses(
+              premisesRunningCosts = expenses.rentsRatesAndInsurance,
+              repairsAndMaintenance = expenses.repairsAndMaintenanceCosts,
+              financialCosts = expenses.loanInterest,
+              professionalFees = expenses.otherProfessionalFee,
+              costOfServices = expenses.costsOfServicesProvided,
+              travelCosts = expenses.propertyBusinessTravelCost,
+              other = expenses.otherAllowablePropertyExpenses,
+              residentialFinancialCostsCarriedForward = None,
```

Comment:
We shouldn't set any value None,  
reason: There could be chance people might use external software to update these fields so lets not override with None

Reply:
Done.